### PR TITLE
Bump scssphp/scssphp to 1.8.1

### PR DESCRIPTION
**Pull request by come-nc**
Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

https://github.com/nextcloud/3rdparty/pull/934

### CODE REVIEW

**Review by CarlSchwan — APPROVED** [review]

**Comment by come-nc** [comment]
```
Error: The method ScssPhp\ScssPhp\Compiler::toHSL is internal to ScssPhp but called from OCA\Theming\Util (see https://psalm.dev/175)
```
Can someone familiar enough look into https://scssphp.github.io/scssphp/docs/changelog.html and see how big the impact of this bump is?

We had 1.4.0 before so it’s a pretty big bump, I’m not sure if the psalm spotted problem is the only one.

**Comment by CarlSchwan** [comment]
I just did some small grepping and the only misused was the one reported by psaml and I fixed it. Other than that locally everything seems to still works and I followed https://scssphp.github.io/scssphp/docs/#security-considerations to do a bit of hardening

File: apps/theming/lib/Util.php [hunk 1]
```diff
@@ -96,15 +95,52 @@ public function elementColor($color, bool $brightBackground = true) {
...
+				$s = $d / $l;
+			} else {
+				$s = $d / (510 - $l);
+			}
+
+			if ($red == $max) {
+				$h = 60 * ($green - $blue) / $d;
+			} elseif ($green == $max) {
+				$h = 60 * ($blue - $red) / $d + 120;
+			} elseif ($blue == $max) {
```
Comment (come-nc) [inline apps/theming/lib/Util.php:128]: Replacing the last `elseif` by `else` should be safe and fix the code scanning `PossiblyUndefinedVariable`

Comment (come-nc) [reply apps/theming/lib/Util.php:128]:
```suggestion
			} else {
```

**Review by skjnldsv — APPROVED** [review]
We can merge but note that this scsscacher should go away as soon as we have the time to :)

**Comment by come-nc** [comment]
@skjnldsv You merged here but it was needed to merge https://github.com/nextcloud/3rdparty/pull/934 first and then rebase here to use the merge commit.

**Comment by ChristophWurst** [comment]
@come-nc make your PR a draft next time :wink:

**Comment by come-nc** [comment]
> @come-nc make your PR a draft next time wink

But can it get approvals while still being a draft? I wait for CI and approvals on server side before merging 3rdparty side, I thought this was not possible with a draft.

**Comment by come-nc** [comment]
(3rdparty was merged, fixing PR server-side incoming)

**Comment by nickvergessen** [comment]
> But can it get approvals while still being a draft?

yes

**Pull request merged**